app/serializers/forest_liana/serializer_factory.rb

```diff
@@ -108,10 +108,14 @@ def relationship_related_link(attribute_name)
 
           if ret[:href].blank?
             begin
-              relationship_records = object.send(attribute_name)
+              if @options[:include].try(:include?, attribute_name.to_s)
+                object.send(attribute_name)
+              end
 
-              if relationship_records.respond_to?(:each)
-                ret[:href] = "/forest/#{object.class.table_name}/#{object.id}/relationships/#{attribute_name}"
+              SchemaUtils.associations(object.class).each do |a|
+                if a.name == attribute_name
+                  ret[:href] = "/forest/#{object.class.table_name}/#{object.id}/relationships/#{attribute_name}"
+                end
```

Comment:
New Regression with the latest commit:
<img width="1259" alt="screen shot 2016-11-03 at 16 28 49" src="https://cloud.githubusercontent.com/assets/664261/19989418/901d50bc-a1e3-11e6-9d71-edc9125e6c5a.png">